### DISCUSSION TITLE
Undo the last character

### DISCUSSION THREAD

**Discussion by MaethorNaur** [8VIM/8VIM#430] [Polls]
From the issue #428.
It has been suggested a feature to "undo" the last character by going in another direction.
This would allow us to correct input mistakes.

However, we would have to take care of layers switching, or other custom movement sequences which require backtracking (those should also have been prioritised over the undo)

The question is would you like having that kind of feature?
Give your thoughts about it.

**Comment by Aiz0** [comment]
I vote no,

I don't think having both sequences that require backtracking and the ability to undo mixes well...
Having the undo sometimes not work would seem weird. (would you have to backtrack the backtracking to undo it?)

Most of my errors come from going into the wrong sector when starting a sequence so I don't need it, but I do need åäö.

I think this might be useful when learning but when you know the layout by feel it's not very useful.
Some ideas on how to make the learning easier. there might be issues already opened for this...
1. Easier way to find learning resources in the app.
2. Show highlighted letter in a position that won't be covered by finger (top right/left for example).
3. option to make cursor movements only trigger on release. (new users might stay in a sector for a while and having it trigger an action they don't want can be bad)

**Reply by MaethorNaur** [reply]
I'm not keen either on an undo feature.

1. We are planning a better way to learn 8vim (through an in-app tutorial/map screen)
2. The UI need indeed a lifting 😄 
3. That's a good idea indeed 👍🏾

**Comment by ChickenF622** [comment]
Could a more complex gesture work, like swiping back and forth? Loving this novel appoach to a keyboard

**Reply by MaethorNaur** [reply]
That might be possible (as long the movement goes from left to right/right to left going through the center
